### DISCUSSION TITLE
Main actor-isolated property error.

### DISCUSSION THREAD

**Discussion by MrCosney** [Quick/Nimble#1055] [Q&A]
Hello!

When I am trying to write test on my custom view in expect block I got the following error:

`Main actor-isolated property 'backgroundColor' can not be referenced from a non-isolated autoclosure`

My Test:

```
override func spec() {
        var view: MyView!
      
        beforeEach {
            view = MyView()
        }
        
        describe("setup") {
            it("should call setup and update background color") {
                view.setup()
                // then
                expect(view.backgroundColor).to(equal(.black)) // here I got an error. But if I use dispatchQueue.main.async for this is okay.
             }
          }   
}             
```

This is my test view:

```
final class MyView: UIView {
    override init(frame: CGRect) {
        super.init(frame: frame)
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func setup() {
        backgroundColor = .black
    }
}
```

- Nimble (11.2.1)
- Quick (6.1.0)

It can't initialise it not on main thread somehow. How to fix it? Thank you!

**Comment by younata** [comment]
You can also use expect with closures, like so:

```swift
// ... the rest of your setup

it("should call setup and update background color") {
    view.setup()
    // then
    await expect { await view.backgroundColor }.to(equal(.black))
}
```

I would also suggest checking out the [Quick 7 beta](https://github.com/Quick/Quick/releases/tag/v7.0.0-beta.3). In Quick 7, `QuickSpec` subclasses will run in synchronous contexts (no async/await) and your original code will work as you expect it to. And if you want to test async methods, you'll need to use `AsyncSpec` subclasses.
